packages/opentelemetry-api/src/context/propagation/setter.ts

```diff
@@ -14,6 +14,18 @@
  * limitations under the License.
  */
 
-export type Carrier = {
-  [key: string]: unknown;
-};
+export type SetterFunction<Carrier = any> = (
+  carrier: Carrier,
+  key: string,
+  value: unknown
```

Comment:
should the value be a string type ?

Reply:
Not necessarily. For things like gRPC it can be numbers and such.